### PR TITLE
Adding further ultraliser deps.

### PR DESCRIPTION
**Pull request by marwan-abdellah**
Ultraliser deps.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ultraliser/package.py [hunk 1]
```diff
@@ -16,3 +16,6 @@ class Ultraliser(CMakePackage):
 
     depends_on('libtiff')
     depends_on('ilmbase')
+    depends_on('vtk')
+    depends_on('hdf5')
+
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/ultraliser/package.py:21]:
```suggestion
```

File: var/spack/repos/builtin/packages/ultraliser/package.py [hunk 2]
```diff
@@ -16,3 +16,6 @@ class Ultraliser(CMakePackage):
 
     depends_on('libtiff')
     depends_on('ilmbase')
+    depends_on('vtk')
+    depends_on('hdf5')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/ultraliser/package.py]:
```suggestion
    depends_on('hdf5+cxx+hl')
```

**Review by matz-e — COMMENTED** [review]
@marwan-abdellah current build fails, as we don't enable the C++ bindings for HDF5 in our deployment. That's something that should actually be added to earlier stages.

File: var/spack/repos/builtin/packages/ultraliser/package.py (hunk 2 above)
Comment (marwan-abdellah) [reply var/spack/repos/builtin/packages/ultraliser/package.py]:
```==> Error: ProcessError: Command exited with status 1:
    '/nvme/abdellah/1213449/spack-stage-hdf5-1.10.5-4csdgbo2bajxnltf5qvvsngureem7dxc/spack-src/configure' '--prefix=/gpfs/bbp.cscs.ch/home/abdellah/software/install/linux-rhel7-x86_64/gcc-8.3.0/hdf5-1.10.5-4csdgb' '--enable-unsupported' '--enable-symbols=yes' '--disable-threadsafe' '--enable-cxx' '--enable-hl' '--disable-fortran' '--without-szlib' '--enable-build-mode=production' '--enable-shared' 'CFLAGS=-fPIC' 'CXXFLAGS=-fPIC' 'FCFLAGS=-fPIC' '--enable-parallel' 'CC=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/hpe-mpi-2.21-7pbszh6v5u/bin/mpicc' 'CXX=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/hpe-mpi-2.21-7pbszh6v5u/bin/mpicxx' '--with-zlib=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/external-libraries/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/zlib-1.2.11-5jk2xjgi67'

2 errors found in build log:
     29    checking for config ./config/site-specific/host-r1i7n26... no
     30    checking build mode... production
     31    checking for gcc... /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/hpe-mpi-2.21-7pbszh6v5u/bin/mpicc
     32    checking whether the C compiler works... yes
     33    checking for C compiler default output file name... a.out
     34    checking for suffix of executables...
  >> 35    checking whether we are cross compiling... configure: error: in `/nvme/abdellah/1213449/spack-stage-hdf5-1.10.5-4csdgbo2bajxnltf5qvvsngureem7dxc/spack-src':
  >> 36    configure: error: cannot run C compiled programs.
     37    If you meant to cross compile, use `--host'.
     38    See `config.log' for more details

```

Comment (matz-e) [reply var/spack/repos/builtin/packages/ultraliser/package.py]: Could you try running this on either one of the login nodes, or purging all `PMI` and `SLURM` related variables from the environment? Our cluster setup messes with things when compiling stuff on compute nodes.

Comment (matz-e) [reply var/spack/repos/builtin/packages/ultraliser/package.py]: We have this in some scripts:
```
unset $(env|awk -F= '/^(PMI|SLURM)_/ {if (match($1, "_(ACCOUNT|PARTITION)$")==0) print $1}')
```

The above removes most MPI related environment variables that mess with compiler detection etc.

Comment (marwan-abdellah) [reply var/spack/repos/builtin/packages/ultraliser/package.py]:
```/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
cd /gpfs/bbp.cscs.ch/project/proj3/development/Ultraliser/spack-build/apps/ultraMeshes2Mesh && /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/cmake-3.15.3-cccsbe7omc/bin/cmake -E copy /gpfs/bbp.cscs.ch/project/proj3/development/Ultraliser/spack-build/bin/ultraMeshes2Mesh /gpfs/bbp.cscs.ch/project/proj3/development/Ultraliser/spack-build/bin
/usr/bin/ld: warning: libHalf.so.6, needed by /usr/lib64/libIlmImf.so, may conflict with libHalf.so.24
/usr/bin/ld: warning: libHalf.so.6, needed by /usr/lib64/libIlmImf.so, may conflict with libHalf.so.24
/usr/bin/ld: warning: libHalf.so.6, needed by /usr/lib64/libIlmImf.so, may conflict with libHalf.so.24
/usr/bin/ld: warning: libHalf.so.6, needed by /usr/lib64/libIlmImf.so, may conflict with libHalf.so.24
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5.so.8, needed by /usr/lib64/libhdf5_cpp.so, may conflict with libhdf5.so.103
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/make[2]: Leaving directory `/gpfs/bbp.cscs.ch/project/proj3/development/Ultraliser/spack-build'
lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
/usr/bin/ld: warning: libhdf5_hl.so.8, needed by /usr/lib64/libhdf5_hl_cpp.so, may conflict with libhdf5_hl.so.100
../../lib/libultraliser.a(VasculatureH5Reader.cpp.o): In function `Ultraliser::VasculatureH5Reader::VasculatureH5Reader(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
VasculatureH5Reader.cpp:(.text+0x3ba): undefined reference to `H5::H5File::H5File(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, H5::FileCreatPropList const&, H5::FileAccPropList const&)'
collect2: error: ld returned 1 exit status

```

**Comment by marwan-abdellah** [comment]
Till the new changes are applied.

**Pull request closed without merging**